src/kademlia/node.cpp

```diff
@@ -966,8 +966,12 @@ void node::incoming_request(msg const& m, entry& e)
 		};
 
 		// attempt to parse the message
+		// also reject the message if it has any non-fatal encoding errors
+		// because put messages contain a signed value they must have correct bencoding
+		// otherwise re-encoding the value will break the signature
```

Comment:
I can certainly do the checking as part of `bdecode`. In fact my original version worked that way. I moved the checking to a separate function so that the added cost would only be incurred when necessary.